netboot-services/ipxeMenuGenerator/menu.ipxe.j2

```diff
@@ -5,16 +5,40 @@
 # Language-Detection: Set different languages based on the default gateway IP address
 :language
 # Lausanne
-iseq ${netX/gateway} 172.20.72.1 && set language fr_CH && goto initial_menu ||
+iseq ${netX/gateway} 172.20.72.1 && set language fr_CH && goto check_boot_from_azure ||
 # Genf
-iseq ${netX/gateway} 172.20.56.1 && set language fr_CH && goto initial_menu ||
+iseq ${netX/gateway} 172.20.56.1 && set language fr_CH && goto check_boot_from_azure ||
 # Krefeld
-iseq ${netX/gateway} 172.22.32.1 && set language de_DE && goto initial_menu ||
+iseq ${netX/gateway} 172.22.32.1 && set language de_DE && goto check_boot_from_azure ||
 # Odilia
-iseq ${netX/gateway} 172.22.224.1 && set language de_DE && goto initial_menu ||
+iseq ${netX/gateway} 172.22.224.1 && set language de_DE && goto check_boot_from_azure ||
 # Fallback (when no condition above triggers to set next-server)
 set language de_CH
 
+:check_boot_from_azure
+iseq ${next-server} {{ azureNetbootServerIP }} && goto register_basic_auth ||
+# if nothing applies, boot from local network
+set http-protocol http && set url ${next-server} && goto initial_menu ||
+
+:register_basic_auth
+set httpAuthUser {{ httpAuthUser }} && set httpAuthPassword {{ httpAuthPassword }} && goto check_if_onprem_netbootserver_is_reachable || 
+goto check_if_onprem_netbootserver_is_reachable ||
+
+:check_if_onprem_netbootserver_is_reachable
+imgfetch https://${httpAuthUser}:${httpAuthPassword}@{{ onpremExposedNetbootServer }}:8443/kernels/latest-kernel-version.json && goto set_onprem_netbootserver ||
+goto check_if_storage_account_is_reachable ||
+
+:check_if_storage_account_is_reachable
```

Comment:
Do I confuse 2 different features now?
In my mind, this should be a check for the reverse proxy.

Reply:
I struggled a bit with the naming here. The value is indeed resolving to the reverseproxy, which redirects traffic to our onprem netbootserver. I'm open for inputs